src/main/java/net/wovenmc/woven/api/item/settings/MeterComponent.java

```diff
@@ -66,27 +63,27 @@ public boolean displayAtFull() {
 	 * A builder for meter components.
 	 */
 	public static class Builder {
-		private Function<ItemStack, Float> levelFunc = stack ->
+		private LevelHandler levelHandler = stack ->
 				(stack.getMaxDamage() - stack.getDamage()) / (float) stack.getMaxDamage();
-		private BiFunction<ItemStack, Float, Integer> colorFunc = (stack, level) ->
-				MathHelper.hsvToRgb(levelFunc.apply(stack) / 3F, 1F, 1F);
+		private ColorHandler colorHandler = (stack, level) ->
+				MathHelper.hsvToRgb(levelHandler.getLevel(stack) / 3F, 1F, 1F);
```

Comment:
Why calling the `levelHandler#getLevel` method here if the value is already given in `level`?

It duplicates executed code.

Reply:
oops